jetty-core/jetty-http3/jetty-http3-server/src/main/java/org/eclipse/jetty/http3/server/internal/HttpStreamOverHTTP3.java

```diff
@@ -133,6 +133,16 @@ private void onBadMessage(HttpException x)
         // TODO
     }
 
+    @Override
+    public void willRead()
+    {
+        if (expects100Continue)
+        {
+            expects100Continue = false;
+            send(requestMetaData, HttpGenerator.CONTINUE_100_INFO, false, null, Callback.NOOP);
```

Comment:
Now this makes me wonder if this isn't racing with a user-generated write? I mean such handler:

```java
boolean handle(Request req, Response resp, Callback callback)
{
  req.demand(() -> {}); // this triggers a write with a NOOP callback
  resp.write(true, BufferUtil.EMPTY, callback); // doesn't this risk a WritePendingException?
}
```

Reply:
I was thinking the same.

Reply:
@review that in the other PR, this PR is based on #10957, so you are seeing both changes here.